File: Binary Search/Valid_Perfect_square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        if(num < 4 && num>1)
            return false;
        else if(num == 1)
            return true;
        else
        {
            long mid;
            int low = 1, high = num;
            while (low <= high) 
            {
                mid = low + (high - low) / 2;
                if (mid*mid == num) 
                    return true;
                else if (mid*mid > num) high = mid-1;
                else if (mid*mid < num) low = mid + 1;
            }
            return false;
        }
        
    }
};

File: Binary Search/Valid_Perfect_square.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        if num < 4 and num > 1:
            return False;
        elif num == 1: return True
        else:
            low = 1
            high = num
            while low <= high:
                mid = low + int((high - low) / 2)
                if mid*mid == num:
                    return True
                elif mid*mid > num : high = mid-1
                elif mid*mid < num : low = mid + 1
            return False